blueye/protocol/udp_protocol_parser.py

```diff
@@ -56,11 +56,12 @@ def unpack_data_dict(self, data):
         return dict(zip(self.get_field_names(packet_type), struct.unpack(struct_format, data)))
 
     def np_array_from_file(self, abspath):
-        with open(abspath, 'rb') as f:
-            version, packet_type = f.read(2)
-        dtype = list(zip(self.get_field_names(packet_type, version=version),
-                         self.get_numpy_field_dtypes(packet_type, version=version)))
-        return np.fromfile(abspath, dtype=dtype)
+        open_ = gzip.open if abspath.endswith(".gz") else open
```

Comment:
Elegant 😍

Reply:
Thanks. Unfortunately, it's not exactly the same interface from the viewpoint of NumPy even though it's supposed to be a file object (likely uses c file descriptors or so). Therefore, I can't give the file descriptor directly to NumPy and need to read it into memory. Could do an if to only do that for zipped files, but as the plan is to mainly use zipped files, I went for less code instead of better performance in case the file isn't zipped.